model/src/main/java/org/projectnessie/model/UriUtil.java

```diff
@@ -44,6 +49,9 @@ private UriUtil() {}
    * @return String encoded for path use.
    */
   public static String toPathString(List<String> elements) {
-    return elements.stream().map(x -> x.replace('.', '\u0000')).collect(Collectors.joining("."));
+    return elements.stream()
+        .map(x -> x.replace(DOT, GROUP_SEPARATOR))
+        .map(x -> x.replace(ZERO_BYTE, GROUP_SEPARATOR))
```

Comment:
Why do we need to replace `ZERO_BYTE`? It's not allowed inside element data, is it?

Reply:
Old clients could still send it

Reply:
I guess defensive programming :)

Reply:
true, but this is the encoding path, how can zero bytes make their way here?